### PR TITLE
Integration tests orphaned shares + Prefilter inaccessible shares

### PR DESCRIPTION
**Pull request by LukasReschke**
Downstreamed from https://github.com/owncloud/core/pull/25977 and https://github.com/owncloud/core/pull/26016

### CODE REVIEW

**Comment by nickvergessen** [comment]
I'd like to have @rullzer reviewing the sharing part.

**Comment by nickvergessen** [comment]
@rullzer since you are back now

**Comment by MorrisJobke** [comment]
The code looks good and also works :+1: 

@rullzer your turn :)

**Review by MorrisJobke — APPROVED** [review]

File: build/integration/features/sharing-v1.feature [hunk 1]
```diff
@@ -906,17 +906,20 @@ Feature: sharing
     And the single response should contain a property "{http://owncloud.org/ns}permissions" with value "SRDNVCK"
     And as "user1" the folder "merge-test-outside-groups-renamebeforesecondshare" does not exist
 
-#  Scenario: Merging shares for recipient when shared from outside with user then group and recipient renames in between
```
Comment (rullzer) [inline build/integration/features/sharing-v1.feature]: leave this for now please. This test someties files I still have no clue why. (It mainly fails on CI).

**Comment by rullzer** [comment]
This tightens the defaultshare provider more to the current file implementation. Which is uhm. less ideal. But I see no current other way to do this without huge performance penalties.

so LGTM!

**Comment by MorrisJobke** [comment]
:+1:

**Pull request merged**